### ISSUE TITLE
cannot install (doesn't exist or not a regular file)

### ISSUE THREAD

**Issue by Lucie-999** [coin-or/Ipopt#343]
Hi, I have a problem installing the package. It ends with an error while building wheel for ipopt.

error: can't copy 'lib_mt\x64\release\Ipopt39.dll': doesn't exist or not a regular file
  ----------------------------------------
  ERROR: Failed building wheel for ipopt

Can you help?

I have no idea where to look fot this missing or not regular file.

Thanks in advance.

Lucie

**Comment by svigerske** [comment]
Ipopt39 sounds very old. We are at 3.12 at the moment.

wheel may have something to do with Python.

I have no idea what you actually did to get to this error. This doesn't look like Ipopt's buildsystem.

**Comment by Lucie-999** [comment]
I tried to install the package 1) by pip install ipopt 2) by manually downloading the files and using python setup.py install

During the building procedure, this error occurs :(

**Comment by svigerske** [comment]
So if I look at https://pypi.org/project/ipopt/, then it looks like you try [cyipopt](https://github.com/matthias-k/cyipopt). You might want to check with their issue tracking. For instance, matthias-k/cyipopt#45 looks similar to your issue.

**Comment by Lucie-999** [comment]
I looked at the link you sent me. In their issue trecking they are referring to ipopt when this issue comes. So I would not think that the problém is with cyipopt…

**Comment by svigerske** [comment]
Well, they essentially say that they don't support Windows.
Nor do I want to spend effort into creating uptodate Ipopt DLLs at the moment.

matthias-k/cyipopt#45 has some write up that seems to suggest using the Windows Subsystem for Linux. That might work.

**Comment by tkralphs** [comment]
The `setup.py` file that you are trying to run seems to assume that an Ipopt DLL has already been built and installed. The path `lib_mt\x64\release\Ipopt39.dll` that is referred to in the error message is hard-coded into `setup.py`. I guess you need to get an Ipopt DLL and then edit `setup.py` to reflect the correct paths.

**Comment by Lucie-999** [comment]
Thanks a lot for the advice. Actually I just tried looking into the setup.py code and apparently in the instalation folder, the files

lib_mt\\x64\\release\\Ipopt39.dll',
   'lib_mt\\x64\\release\\IpoptFSS39.dll

are missing. When I added such files manually, the instalation seems to have finished successfully. The question is whether or not I managed to get the correct files – I did not find those in here…
